### PR TITLE
Update theforeman dev dependencies to version ^10.0

### PR DESCRIPTION
**Pull request by ekohl**
These are the latest versions. Packaging already uses these versions.

I have no clue if this works, just submitting it to see what CI thinks. This what RPMs run with in production so if it fails here, it will probably also fail in production.

### CODE REVIEW

**Comment by ekohl** [comment]
Derp, it was `< 11` in our packaging and I copied that. Updated to 10.

**Comment by ekohl** [comment]
Ruby tests fail due to Rails 6.1.6.1.

File: package.json [hunk 1]
```diff
@@ -15,12 +15,12 @@
...
-    "@theforeman/find-foreman": "^8.4.1",
-    "@theforeman/stories": "^8.4.1",
-    "@theforeman/test": "^8.9.0",
-    "@theforeman/vendor-dev": "^8.4.1",
+    "@theforeman/builder": "^10.0",
+    "@theforeman/eslint-plugin-foreman": "^10.0",
+    "@theforeman/find-foreman": "^10.0",
+    "@theforeman/stories": "^10.0",
+    "@theforeman/test": "^10.0",
+    "@theforeman/vendor-dev": "^10.0",
```
Comment (Ron-Lavi) [inline package.json:23]: On other plugins we started to accept `>= 0` for these versions, to reduce the number of changes for plugins:
```suggestion
    "@theforeman/builder": ">= 0",
    "@theforeman/eslint-plugin-foreman": ">= 0",
    "@theforeman/find-foreman": ">= 0",
    "@theforeman/stories": ">= 0",
    "@theforeman/test": ">= 0",
```
btw, `@theforeman/vendor-dev` isn't needed.

same for the peerDeps:
```json
"peerDependencies": {
  "@theforeman/vendor": ">= 0"
},
```

reference: https://github.com/theforeman/foreman_discovery/blob/develop/package.json#L28

Comment (ekohl) [reply package.json:23]: I don't have time to finish this up, but this is blocking updates. Could you have a look @Ron-Lavi?

Comment (Ron-Lavi) [reply package.json:23]: Hey Ewoud, I am actually on vacation and can't test it, @mariaaga do you mind taking a look? Thanks

Comment (ekohl) [reply package.json:23]: Then by all means, enjoy your vacation. No need to reply immediately :)

**Review by MariaAga — APPROVED** [review]
I'm against using `">= 0"`, so this lgtm. only breaking change between the versions was related to `react-ace` (and its tests) which foreman_ansible doesnt use.

**Review by Ron-Lavi — APPROVED** [review]
Thanks @ekohl, we can add the suggestions on a follow-up

**Pull request merged**